packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -31,3 +32,10 @@ class _WindowsUtils implements OperatingSystemUtils {
     return new ProcessResult(0, 0, null, null);
   }
 }
+
+Future<int> findAvailablePort() async {
+  ServerSocket socket = await ServerSocket.bind(InternetAddress.LOOPBACK_IP_V4, 0);
+  int port = socket.port;
+  await socket.close();
+  return port;
```

Comment:
It'll pick one on the device - we need one for clients to connect to on the host.